### DISCUSSION TITLE
Bluetooth BMS install stalled

### DISCUSSION THREAD

**Discussion by rldoose** [Louisvdw/dbus-serialbattery#860] [Q&A]
I'm attempting to get a BLE BMS setup for the first time.  File  /data/etc/dbus-serialbattery/config.ini reads as follows.

[DEFAULT]
 MAX_BATTERY_CHARGE_CURRENT = 50.0
 MAX_BATTERY_DISCHARGE_CURRENT = 60.0
MIN_CELL_VOLTAGE   = 2.5
MAX_CELL_VOLTAGE   = 3.650
FLOAT_CELL_VOLTAGE = 3.375
BLUETOOTH_BMS = LltJbd_Ble C3:27:20:37:C1:A4

After running /data/etc/dbus-serialbattery/reinstall-local.sh, the install gets to the state below:
Found 1 Bluetooth BMS in the config file!

Installing required packages...
Downloading https://updates.victronenergy.com/feeds/venus/release/packages/dunfell/all/Packages.gz.
Updated source 'all'.
Downloading https://updates.victronenergy.com/feeds/venus/release/packages/dunfell/cortexa7hf-neon-vfpv4/Packages.gz.
Updated source 'cortexa7hf-neon-vfpv4'.
Downloading https://updates.victronenergy.com/feeds/venus/release/packages/dunfell/raspberrypi2/Packages.gz.
Updated source 'raspberrypi2'.
Installing python3-xmlrpc (3.8.14) on root
Downloading https://updates.victronenergy.com/feeds/venus/release/packages/dunfell/cortexa7hf-neon-vfpv4/python3-xmlrpc_3.8.14-r0_cortexa7hf-neon-vfpv4.ipk.
Installing libjitterentropy2 (2.2.0) on root
Downloading https://updates.victronenergy.com/feeds/venus/release/packages/dunfell/cortexa7hf-neon-vfpv4/libjitterentropy2_2.2.0-r0_cortexa7hf-neon-vfpv4.ipk.
Installing libsysfs2 (2.1.0) on root
Downloading https://updates.victronenergy.com/feeds/venus/release/packages/dunfell/cortexa7hf-neon-vfpv4/libsysfs2_2.1.0-r5_cortexa7hf-neon-vfpv4.ipk.
Installing python3-pip (20.0.2) on root
Downloading https://updates.victronenergy.com/feeds/venus/release/packages/dunfell/cortexa7hf-neon-vfpv4/python3-pip_20.0.2-r0_cortexa7hf-neon-vfpv4.ipk.
Installing rng-tools (6.9) on root
Downloading https://updates.victronenergy.com/feeds/venus/release/packages/dunfell/cortexa7hf-neon-vfpv4/rng-tools_6.9-r0_cortexa7hf-neon-vfpv4.ipk.
Configuring libjitterentropy2.
Configuring libsysfs2.
Configuring rng-tools.
 Adding system startup for /etc/init.d/rng-tools.
Starting random number generator daemon.
Configuring python3-xmlrpc.
Configuring python3-pip.
Collecting bleak
  Downloading bleak-0.21.1-py3-none-any.whl (137 kB)
     |████████████████████████████████| 137 kB 286 kB/s 
Collecting typing-extensions>=4.7.0; python_version < "3.12"
  Downloading typing_extensions-4.8.0-py3-none-any.whl (31 kB)
Collecting dbus-fast<3,>=1.83.0; platform_system == "Linux"
  Downloading dbus_fast-2.14.0.tar.gz (68 kB)
     |████████████████████████████████| 68 kB 289 kB/s 
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... done
Collecting async-timeout<5,>=3.0.0; python_version < "3.11"
  Downloading async_timeout-4.0.3-py3-none-any.whl (5.7 kB)
Building wheels for collected packages: dbus-fast
  Building wheel for dbus-fast (PEP 517) ... -

Any idea why it is hanging?
Thanks,
Reagan

**Comment by mr-manuel** [comment]
By looking at my crystal ball I assume that you are trying to install v1.0.20230531 and you are trying to install it on a weaker device like a Cerbo :-D This is a know issue and is fixed in the latest nightly of the dev branch.

Please read also the pinned discussion https://github.com/Louisvdw/dbus-serialbattery/discussions/856

**Comment by rldoose** [comment]
I really tried hard to meticulously fill your form.  Apologies if I missed the  version.  I do know that Raspberry Pi was selected as my hardware.  And, as I stated, I tried it on both RPi 3&4.

**Reply by mr-manuel** [reply]
Sorry but you did not mention, that you opened also a issue https://github.com/Louisvdw/dbus-serialbattery/issues/861. In this thread the info is missing. If you don't link it, we cannot know all.

**Comment by rldoose** [comment]
My apologies.  Thought I was just asking for community ideas.  Did not realize this one would end up a support request.

**Reply by mr-manuel** [reply]
Did you try the nightly?

**Comment by rldoose** [comment]
I have it installed, but need to connect it to battery.  Under transit in
RV, so no Wi-Fi to RPi.  Will setup Starlink tonight and try then.

Thanks for checking!
Reagan

**Comment by rldoose** [comment]
I tried the nightly build.  No initial success, but before I do further troubleshooting, I have a question.  If “svstat /service/dbus-serialbattery.tty*” indicates the the driver is changing the PID about every 60 seconds, indicating “your driver is not stable and has a problem”, then something is wrong and must be fixed before further troubleshooting correct?   

BTW, I’m now back to trying the serial connection and have #861 open on that.

**Reply by mr-manuel** [reply]
Just check the logs instead of svstat :-)
